kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/DerivedResourcesTest.java

```diff
@@ -201,99 +169,160 @@ static List<Path> filesInDir(Path dir, String glob) {
+                                                 Path testDir)
+            throws IOException {
+        var unusedFiles = childFilesMatching(testDir, "*");
+        Path input = testDir.resolve("in-KafkaProxy.yaml");
+        KafkaProxy kafkaProxy = kafkaProxyFromFile(input);
+        unusedFiles.remove(input);
+        unusedFiles.remove(testDir.resolve("operator-config.yaml"));
+        unusedFiles.removeAll(childFilesMatching(testDir, "in-*"));
+
+        Context<KafkaProxy> context;
+        try {
+            context = buildContext(kafkaProxy, testDir);
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
 
         List<DynamicTest> tests = new ArrayList<>();
 
-        var allPresentExpectedFilenames = filesInDir(testDir, "out-*").stream()
-                .map(testDir::relativize)
+        var dr = dependentResources.stream()
+                .flatMap(r -> r.invokeDesired(kafkaProxy, context).values().stream().map(x -> Map.entry(r.resourceType(), x)))
+                .collect(Collectors.groupingBy(Map.Entry::getKey))
+                .entrySet()
+                .stream()
+                .collect(Collectors.toMap(Map.Entry::getKey,
+                        e -> e.getValue().stream().map(Map.Entry::getValue).collect(Collectors.toCollection(() -> new TreeSet<>(
+                                Comparator.comparing(hasMetadata -> hasMetadata.getMetadata().getName()))))));
+        for (var entry : dr.entrySet()) {
+            var resourceType = entry.getKey();
+            var actualResources = entry.getValue();
+            for (var actualResource : actualResources) {
+                String kind = resourceType.getSimpleName();
+                String name = actualResource.getMetadata().getName();
+                var expectedFile = testDir.resolve("out-" + kind + "-" + name + ".yaml");
+                tests.add(DynamicTest.dynamicTest(kind + " '" + name + "' should have the same content as " + testDir.relativize(expectedFile),
+                        () -> {
+                            assertThat(Files.exists(expectedFile)).isTrue();
+                            var expected = loadExpected(expectedFile, resourceType);
+                            assertSameYaml(actualResource, expected);
+                            unusedFiles.remove(expectedFile);
+                        }));
+            }
+            for (var cluster : kafkaProxy.getSpec().getClusters()) {
+                ClusterCondition actualClusterCondition = SharedKafkaProxyContext.clusterCondition(context, cluster);
+                if (actualClusterCondition.type() == ConditionType.Accepted && actualClusterCondition.status().equals(Conditions.Status.TRUE)) {
+                    continue;
+                }
+                else {
+                    var expectedFile = testDir.resolve("cond-" + actualClusterCondition.type() + "-" + actualClusterCondition.cluster() + ".yaml");
```

Comment:
I like the `in-/out-` prefixing on the other files. Would be cool if this could have a common prefix like `out-cond` (but I can see it messes with the existing logic). Something to improve later.

Reply:
FWIW, I didn't use `out` because the way it currently works the conditions are not returned from the `desired()` methods, but a side-effect on the `context`.